pkg/apiserver/slowquery/queries.go

```diff
@@ -114,37 +115,13 @@ type GetDetailRequest struct {
 	ConnectID int64   `json:"connect_id" form:"connect_id"`
 }
 
-func getAllColumnNames() []string {
-	t := reflect.TypeOf(Base{})
-	fieldsNum := t.NumField()
-	ret := make([]string, 0, fieldsNum)
-	for i := 0; i < fieldsNum; i++ {
-		field := t.Field(i)
-		if s, ok := field.Tag.Lookup("gorm"); ok {
-			list := strings.Split(s, ":")
-			if len(list) < 1 {
-				panic(fmt.Sprintf("Unknown gorm tag field: %s", s))
-			}
-			ret = append(ret, list[1])
-		}
-	}
-	ret = append(ret, "Time")
-	return ret
-}
-
-func isValidColumnName(name string) bool {
-	for _, item := range getAllColumnNames() {
-		if name == item {
-			return true
-		}
-	}
-	return false
-}
-
-func QuerySlowLogList(db *gorm.DB, req *GetListRequest) ([]Base, error) {
+func QuerySlowLogList(db *gorm.DB, req *GetListRequest) ([]SlowQuery, error) {
+	finalFields := strings.TrimRight(
+		"Digest,Conn_id,Time,(unix_timestamp(Time) + 0E0) as timestamp,"+strings.TrimSpace(req.Fields),
```

Comment:
you must not pass req.Fields directly since it will lead to SQL injections

Reply:
Got, will update it soon.